Shopify/cli
# Add  app draft-extensions push command

### WHY are these changes introduced?
This is the `app draft-extensions push` command implementation explained in this [doc](https://docs.google.com/document/d/1J2_p4qIEQpusqX3_No8vLUvT4_ngXo-tzpgPHnMWVtM/edit#heading=h.yeei0eme1l1p) 



### WHAT is this pull request doing?
- Creates a new hidden command `app draft-extensions push`
- Builds the app extensions and updates their draft content
- Enable dev preview mode when the proper command flag is set



### How to test your changes?
- Run the following command using an app with a `UI extension`, a `theme app extension`,  `js funcion`, `rust function`
```sh
true | (SHOPIFY_CLI_PARTNERS_TOKEN=<YOUR_TOKE> SHOPIFY_FLAG_CLIENT_ID=<APP-CLIENT-ID> pnpm shopify app draft-extensions push --path /your/app/path --enable-dev-preview) 2>&1 | cat
```
- Check in the partners dashboard, in the `extensions` section inside the app that the content of the extensions are updated.


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
